docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -2641,291 +2641,213 @@ The following error message codes may appear on the app-server log. Use the foll
 
 
 
-### ZWED0017E":"Identifier doesn't match one found in pluginDefinition: %s
-
-  **Reason:**
-
-  TODO
-
-  **Action:**
-
-  TODO
+### ZWED0017E 
 
-
-
-### ZWED0018E":"No plugin type found, skipping
+  Identifier doesn't match one found in pluginDefinition: _pluginIdentifier_
 
   **Reason:**
 
-  TODO
+  The identifier found in the plugin reference doesn't match the one specified in the pluginDefinition.json
 
   **Action:**
 
-  TODO
-
-
-
-### ZWED0019E":"Plugin already registered
-
-  **Reason:**
-
-  TODO
-
-  **Action:**
+  Check if identifier found is the same one as intended (typo perhaps?). If not, edit pluginDefinition.json or turn on App server debugging for the bootstrap logger (_zsf.bootstrap) and check if the plugin reference found in code ZWED0044I is correct
 
-  TODO
 
 
+### ZWED0018E 
 
-### ZWED0020E":"%s: pluginType %s is unknown
+  No plugin type found, skipping
 
   **Reason:**
 
-  TODO
+  The plugin definition for the plugin has no 'pluginType' property set
 
   **Action:**
 
-  TODO
-
-
-
-### ZWED0021E":"%s is missing
-
-  **Reason:**
+  Contact app developers if you need plugin to be loaded and working
 
-  TODO
 
-  **Action:**
 
-  TODO
+### ZWED0019E 
 
-
-
-### ZWED0022E":"Module not found %s
+  Plugin already registered
 
   **Reason:**
 
-  TODO
+  A plugin with this identifier has already been registered to the map of plugins
 
   **Action:**
 
-  TODO
-
-
-
-### ZWED0023E":"Method not implemented %s
-
-  **Reason:**
-
-  TODO
-
-  **Action:**
+  Check if you have multiple components sharing the same, or different versions, of the same plugin. This is not allowed
 
-  TODO
 
 
+### ZWED0020E 
 
-### ZWED0024E":"Object not exported %s
+  "_pluginIdentifier_: pluginType _type_ is unknown
 
   **Reason:**
 
-  TODO
+  The plugin _pluginIdentifier_ has in its plugin definition an invalid plugin _type_
 
   **Action:**
 
-  TODO
+  Accepted plugin types: library, application, windowManager, bootstrap, desktop, nodeAuthentication, proxyConnector
 
 
 
-### ZWED0025E":".authenticate() missing
-
-  **Reason:**
-
-  TODO
-
-  **Action:**
-
-  TODO
+### ZWED0021E
 
-
-
-### ZWED0026E":"Circular dependency: %s
+  _pluginPath_ is missing
 
   **Reason:**
 
-  TODO
+  App server tried to process the plugin reference from path _pluginPath_
 
   **Action:**
 
-  TODO
+  Check if _pluginPath_ is a real path
```

Comment:
or if permissions are preventing access? or contact support? unsure about this one.